cmake/findDependencies.cmake

```diff
@@ -59,3 +59,30 @@ if( BUILD_WITH_CCACHE )
     endif()
 endif()
 
+set(CMAKE_REQUIRED_FLAGS "-std=c++11")
+# source: https://stackoverflow.com/questions/12530406/is-gcc-4-8-or-earlier-buggy-about-regular-expressions/41186162#41186162
+check_cxx_source_compiles(
+  "#include <regex>
+#if __cplusplus >= 201103L &&                             \
+    (!defined(__GLIBCXX__) || (__cplusplus >= 201402L) || \
+        (defined(_GLIBCXX_REGEX_DFS_QUANTIFIERS_LIMIT) || \
+         defined(_GLIBCXX_REGEX_STATE_LIMIT)           || \
+             (defined(_GLIBCXX_RELEASE)                && \
+             _GLIBCXX_RELEASE > 4)))
+int main() { return 0; }
+#else
+#error \"regex not working\"
+#endif
+"
+  BUILTIN_REGEX_WORKING
+  FAIL_REGEX '.*regex not working.*'
+  )
+set(CMAKE_REQUIRED_FLAGS "")
+
+if( NOT BUILTIN_REGEX_WORKING )
+  set( EXV_NEED_BOOST_REGEX ON
+    CACHE
+    BOOL
+    "Need Boost::regex as this version of gcc ships a broken implementation of <regex>")
+  find_package(Boost REQUIRED COMPONENTS regex)
```

Comment:
Probably we should specify here the minimum version required.

Reply:
Hm, to be honest I wouldn't know which version to choose for the minimum. I could insert the currently available one from EPEL, but no idea if older ones would work too?